Add OSD alarms for distance and negative altitude. Refactor alarms to use SI units internally.

fiam opened the pull request:
Distance alarm makes the OSD blink when distance to home goes
above a given threshold. Default is 1000m and it always works
on meters to avoid performing unnecessary arithmetic operations
for displaying the distance indicator. Unit conversion is done
in the configurator instead. Zero disables the alarm.

Negative altitude alarm blinks when negative altitude is below
the given threshold. Note that the alarm is a positive number,
but will trigger with -ALT > NEG_ALT_ALARM. Default value is 5
(which means -5 meters). Again, conversion is pushed to the
configurator.

Fixes #2242 and #2199

Configurator PR at https://github.com/iNavFlight/inav-configurator/pull/276

fiam committed:
Add OSD alarms for distance and negative altitude

Distance alarm makes the OSD blink when distance to home goes
above a given threshold. Default is 1000m and it always works
on meters to avoid performing unnecessary arithmetic operations
for displaying the distance indicator. Unit conversion is done
in the configurator instead. Zero disables the alarm.

Negative altitude alarm blinks when negative altitude is below
the given threshold. Note that the alarm is a positive number,
but will trigger with -ALT > NEG_ALT_ALARM. Default value is 5
(which means -5 meters). Again, conversion is pushed to the
configurator.

Fixes #2242 and #2199

digitalentity commented on src/main/io/osd.c:
```diff
@@ -916,7 +922,9 @@ static bool osdDrawSingleElement(uint8_t item)
             osdFormatAltitudeSymbol(buff, alt);
-            if ((osdConvertDistanceToUnit(alt) / 100) >= osdConfig()->alt_alarm) {
+            uint16_t neg_alt_alarm = osdConfig()->neg_alt_alarm;
+            if ((osdConvertDistanceToUnit(alt) / 100) >= osdConfig()->alt_alarm ||
```
This still converts between metric/imperial. Isnt it inconsistent to use different units for some alarms and use only SI units for other alarms?

fiam replied:
It’s converting for the already existing altitude alarm. Didn’t want to change that without asking first. See #2297. If you agree, I can push another commit on top of this PR changing alt_alarm too.

fiam committed:
Convert OSD's alt_alarm to work always in meters

Just like neg_alt_alarm

fiam committed:
Document OSD alarm fields in osd.h

Makes it more clear than documenting in when setting their
defaults in osd.c

fiam committed:
Disable OSD cap_alarm when it's zero, set default value to disabled

Since we support a wide range of machines, there's no sensible value
for a mah used alarm. Is better to leave it as zero and let the
user configure it.
